### DISCUSSION TITLE
Plans to support global workflows in .github repository

### DISCUSSION THREAD

**Discussion by derberg** [community/community#24998] [Actions]
<p>I just found out about this awesome GitHub feature <a href="https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file" rel="nofollow noopener">https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file</a></p>
<p>I immediately noticed that I have few workflows that are exactly the same for the whole organization, like handling stale PRs and Issues.</p>
<p>Did you think about adding support for GitHub Actions in <code>.github</code> repository. Would be even nifty if in such a global workflow file I could define exactly what repositories is it for.</p>
<p>What do you think?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/derberg">@derberg</a> ,</p>
<p>Thank you for reaching this out! Currently github workflow only works per repository.</p>
<p>According to the policy, you are welcome to share your idea <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github product manager will take a review.</p>
<p>Thanks.</p>

**Comment by derberg** [comment]
<p>Thanks a lot for a hint. Sorry for posting it in a wrong location. I just sent a feature proposal. Thank you again</p>

**Comment by ibakshay** [comment]
<p>hi <a class="mention" href="/derberg">@derberg</a>, I have read in one of the discussion that GitHub Actions development team is working on enabling workflows on organization level.  </p>

**Comment by iDVB** [comment]
<p><a class="mention" href="/ibakshay">@ibakshay</a> Can you post the discussion where you saw this?</p>

**Comment by derberg** [comment, marked as answer]
<p>so, I was tired of waiting and could not imagine editing the same workflows in 37 repos manually, so I created a GitHub Action that supports exactly this, global workflows</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/marketplace/actions/global-workflows-support" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6995927/181088055-c4d03159-115a-4fde-a6b2-5d608b6759dd.jpeg" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/marketplace/actions/global-workflows-support" target="_blank" rel="noopener nofollow ugc">Global Workflows Support - GitHub Marketplace</a></h3>

<p>This action introduces support for global workflows</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>you can see it running with success <a href="https://github.com/asyncapi/.github/runs/1463225247?check_suite_focus=true" rel="noopener nofollow ugc">here</a> after <a href="https://github.com/asyncapi/.github/pull/15" rel="noopener nofollow ugc">this</a> PR was merged. PR with updated workflow showed in 37 other repos in less than 3min</p>

**Comment by dskvr** [comment]
<p>This feature would be a huge improvement to GitHub Actions. If this feature ever makes the cut, I have one request</p>
<ol>
<li>
<code>glob</code> or <code>regex</code> matching against a repository name for workflow triggers. This would provide the repository maintainers the ability to match against a self-defined, standardized naming scheme. The aim would be to reduce maintenance requirements through self-defined conventions enabled by non-opinionated configuration.</li>
</ol>
<pre><code class="lang-auto">on: 
  repository: 
    pattern: plugin-* 
</code></pre>
<p>This would be in addition to the more obvious feature, which would be an array of explicitly defined repos</p>
<pre><code class="lang-auto">on: 
  repository: 
    list: [ plugin-a, plugin-b ] 
</code></pre>
<p>Obviously the formatting above is speculative as it would likely be implemented much differently, but it’s my hope that it conveys the idea succinctly</p>

**Comment by derberg** [comment]
<p>did you try my action as a workaround?</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/marketplace/actions/global-workflows-support" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars1.githubusercontent.com/u/6995927?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/marketplace/actions/global-workflows-support" target="_blank" rel="noopener nofollow ugc">Global Workflows Support - GitHub Marketplace</a></h3>


  <p><span class="label1">This action introduces support for global workflows</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I’m using it here <a href="https://github.com/asyncapi/.github/blob/1f1d59b6d59371159cca12960c60e4cf24c433fe/.github/workflows/global-workflows-support.yml" class="inline-onebox" rel="noopener nofollow ugc">.github/global-workflows-support.yml at 1f1d59b6d59371159cca12960c60e4cf24c433fe · asyncapi/.github · GitHub</a> and so far I’m super happy with it</p>

**Comment by hermanbanken** [comment]
<p>Nice workaround. However, I’d really would like GitHub to support this natively!</p>
<p>Our security policy is strict, and running GitHub Actions like this one is not OK. It requires a token that can write to the whole org, and consequently creates commits in all repositories. This would trigger so many micro service deployments in our case, that we can’t use this.</p>
<p>A GitHub native implementation hopefully does not create commits: individual repo’s would just inherit the global workflow for the next PR/push.</p>

**Comment by derberg** [comment]
<p>Definitely. I would be more than happy to throw away my action. Even a day after writing it, but yeah, what else we can do, going to each repo and doing changes manually repo-by-repo would be just crazy</p>

**Comment by Gibby** [comment]
<p><a class="mention" href="/hermanbanken">@hermanbanken</a> if you follow <a href="https://www.conventionalcommits.org/en/v1.0.0/" class="inline-onebox" rel="noopener nofollow ugc">Conventional Commits</a> and update commit_message for this action along the lines of <code>ci: Update global workflows</code>  you could easily exclude that in your microservices deployments.</p>

**Comment by odahcam** [comment]
This should be a feature request, it is so annoying to not be able to configure a global workflow for every issue added to a repository, so you have to go to each repo and configure what should happen to the issue. That's ok, but imagine having dozens of repos and having to configure the same action for all of them.
